gage_streamorder/gage-streamorder.R

```diff
@@ -0,0 +1,47 @@
+# get streamorder of a single gage site  ---------------------
+library(dataRetrieval)
+library(tidyverse)
+library(nhdplusTools)
+library(sf)
+
+# for all gages (as of 2019) - thesite_no, number of years active, when started, and any inactive years
+gages <- readRDS("data/active_flow_gages_summary_wy.rds")
+str(gages)
+
+# make a list of all gages
+gage_list <- gages %>% pull(site) %>% unique
+
+## find drainage area from NWIS
+readNWISsite(gage_list[1]) # for a single site
+gage_da <- readNWISsite(gage_list) # for a all sites
+
+# Pick a random single gage!
+rand <- 10000
+
+# grab coords of a single gage and convert to comid for NHDPLus
+start <- gage_da %>% slice(rand)
+start_point <- st_sfc(st_point(c(start$dec_long_va,start$dec_lat_va)), crs = 4269)
+comid <- discover_nhdplus_id(start_point)
+
+# get the next closest comid within 10 km
+nldi_feature <- list(featureSource = "comid", featureID = start_comid)
+flowline_nldi <- navigate_nldi(nldi_feature, 
+                               mode = "upstreamTributaries", 
+                               distance_km = 100)
+next_comid <- flowline_nldi$UT_flowlines %>% slice(1)
+
+# assign to a variable
```

Comment:
I think these lines will be unnecessary per our conversation about stream order. Alternatively, another function that takes the comid and runs the lines below will download nhd data for each comid, including a variable called `stream_orde`